### ISSUE TITLE
Variables getting inadvertently updated

### ISSUE THREAD

**Issue by phylll** [michael-buschbeck/mychs-macro-magic#210]
Was there a recent change in MMM that gets variables auto-updated somehow?

I've just noticed a weird bug that seems to be tracable to this new behavior:

Code:
```
!mmm script
!mmm   set tokenID = selected
!mmm   set prvHealth = tokenID.LP
!mmm   debug do prvHealth
!mmm   if setattr(tokenID, "LP", 13) == 13
!mmm     debug do prvHealth
!mmm   end if
!mmm end script 
```

Output:
> 2:50PM (From Mych's Macro Magic):
> 🔎 "18" ◀️ prvHealth
> 2:50PM (From Mych's Macro Magic):
> 🔎 13 ◀️ prvHealth

So I've changed `tokenID.LP` from `18` to `13` and as a result, a completely different variable that happened to be initially set to `tokenID.LP` gets updated, as well?

**Comment by michael-buschbeck** [comment]
I have a suspicion:

The `foo.LP` syntax doesn't directly return the requested value but instead a wrapper object that will coerce to the requested value when used directly, but will also support the `foo.LP.max` property to get the attribute max value.

I may have unreflectedly implemented this in such a way that these attributes – `LP` itself or its max value, respectively – are queried lazily right when they need to be coerced to scalar. So `foo.LP` doesn't actually contain `foo`'s current `LP` – it just encapsulates a _potential query_ for `foo`'s `LP`.

That's idiotic and I'd consider it a bug.

**Comment by michael-buschbeck** [comment]
This is fixed in the current 1.35.0 pre-release. (It was nominally fixed as 1.34.1.)
